inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,299 @@
+
+        elif new_schema['type'] == 'array':
+            if new_schema['items']['type'] == 'object':
+                new_schema = new_schema['items']['properties']
+            if len(path) == 1:
+                temp_record[key] = [value]
+                return record
+            else:
+                temp_record[key] = [{}]
+                temp_record = temp_record[key][0]
+    temp_record[path[-1]] = value
+    return record
+
+
+def check_value(record, match_type, keys, value_to_check, position):
+    """Where continues to find the value."""
+    key = keys[position]
+    if not record.get(key):
+        if match_type == 'does not exist':
+            return True
+        else:
+            return False
+    temp_record = record[key]
+    if isinstance(temp_record, list):
+        for index, array_record in enumerate(temp_record):
+            if position + 1 == len(keys):
+                if match_type == 'is equal to' and\
+                        array_record == value_to_check:
+                    return True
+                elif match_type == 'contains' and\
+                        value_to_check in array_record:
+                    return True
+                elif match_type == 'matches regular expression'and\
+                        re.search(
+                            re.escape(value_to_check),
+                            array_record):
+                            return True
+                elif match_type == 'does not exist' and\
+                        value_to_check == array_record:
+                    return True
+            else:
+                if check_value(array_record, match_type,
+                               keys, value_to_check, position + 1):
+                    return True
+    else:
+        if position + 1 == len(keys):
+            if match_type == 'is equal to' and \
+                    temp_record == value_to_check:
+                return True
+            elif match_type == 'contains' and \
+                    value_to_check in temp_record:
+                return True
+            elif match_type == 'matches regular expression' and \
+                    re.search(
+                        re.escape(value_to_check),
+                        temp_record):
+                return True
+        else:
+            return check_value(temp_record, match_type,
+                               keys, value_to_check, position + 1)
+    return False
+
+
+def get_actions(user_actions):
+    class_actions = []
+    where_actions = []
+    if not user_actions:
+        return
+
+    if user_actions.get('conditions'):
+        for action in user_actions.get('conditions'):
+            if not action['key']:
+                continue
+            key = action['key'].split('/')
+            where_action = {'value': action['value'],
+                            'key': key,
+                            'match_type': action['matchType']}
+            where_actions.append(where_action)
```

Comment:
The fact than on one side it is called `conditions` and then `where_actions` can be confusing, can the naming be unified?